lib/ravelin/event.rb

```diff
@@ -0,0 +1,119 @@
+module Ravelin
+  class Event
+    attr_accessor :name, :timestamp, :payload
+
+    def initialize(name:, payload:, timestamp: nil)
+      @name = name.to_sym
+      @payload = convert_to_ravelin_objects(payload)
+      @timestamp = timestamp.nil? ? Time.now.to_i : convert_to_epoch(timestamp)
+
+      validate_top_level_payload_params
+    end
+
+    def serialize
```

Comment:
Decided to go with `#serializable_hash` after our discussion due to the method changing key names to camel-case.